src/common/database.c

```diff
@@ -2524,6 +2533,122 @@ gboolean dt_database_get_lock_acquired(const dt_database_t *db)
   return db->lock_acquired;
 }
 
+void _dt_database_vacuum(const struct dt_database_t *db)
```

Comment:
vaccum does not seem a good name here as it also does optimization (ANALYSE is about this). We can't use optimize which is use for "PRAMA optimize" in dt_database_optimize(). Maybe use _dt_database_reform() or _dt_database_maintenance() ?